src/roaring.c

```diff
@@ -129,6 +129,47 @@ void roaring_bitmap_add_many(roaring_bitmap_t *r, size_t n_args,
     }
 }
 
+void roaring_bitmap_contains_multi(roaring_bitmap_t *r, size_t n_args,
+                             const uint32_t *vals, bool *results) {
+    if (n_args == 0 || r == NULL)  {
+        return;
+    }
+    uint32_t prev_container_key = 0;   // previous container key
+    void *container = NULL;     // hold value of last container touched
+    uint8_t typecode = 0;       // typecode of last container touched
+    int32_t container_index = -1;    // current container pos
+    const int container_count= r->high_low_container.size;
+
+    for (size_t i = 0; i < n_args; i++) {
+        if ((container != NULL) && (((prev_container_key^ vals[i]) >> 16) == 0)) {
+            // currently still in the same container
+            results[i] = container_contains(container, vals[i] & 0xFFFF, typecode);
+        } else {
+            if (container_index != -1 && vals[i] < (prev_container_key & 0xFFFF0000)) {
```

Comment:
Simply assigning every time the second condition is true is both simpler and probably more performant (branches are more expensive than assigning a constant in modern processors):
```suggestion
            if (vals[i] < (prev_container_key & 0xFFFF0000)) {
```